onnxruntime/core/optimizer/conv_add_act_fusion.cc

```diff
@@ -293,6 +295,10 @@ void RegisterConvAddActivationFusionRules(SelectorActionRegistry& registry) {
   auto selector_nhwc = std::make_unique<selectors::ConvAddActivationSelector>();
   registry.RegisterSelectorAndAction("NhwcFusedConvAct", {{"NhwcFusedConv", {1, 11}}},
                                      std::move(selector_nhwc), std::move(action_nhwc));
+  auto action_ms_internal_nhwc = std::make_unique<actions::FuseConvAddActivationAction>(kMSInternalNHWCDomain);
+  auto selector_ms_internal_nhwc = std::make_unique<selectors::ConvAddActivationSelector>();
```

Comment:
I think the existing code is providing a bad example here. There's no need for ConvAddAct and NhwcFusedConvAct to be separate registrations. The action already handles both usages internally so we could combine them with 

```
  registry.RegisterSelectorAndAction("ConvAddAct", {{"Conv", {1, 11}}, {"NhwcFusedConv", {1, 11}}},
                                     std::move(selector), std::move(action));
```

And that single registration could also be used to add support for the internal NHWC domain by 
- applying any EP specific filtering in the Select when the Conv is in the internal NHWC domain, and 
- updating FuseConvAddActivationAction to check the optype and domain of the target node in OpType() and Domain(). the rules are something like this aren't they?  
  - onnx Conv -> kMsDomain FusedConv
  - internal NHWC Conv -> internal NHWC Conv
  - NhwcFusedConv -> NhwcFusedConv

Reply:
Yes. I figure the first part in this comment already. That is how added ConvTranspose in conv_activation_fusion.cc using single registration.